File: app/src/main/java/com/example/retrofitdemoapp/UsersViewModel.java
package com.example.retrofitdemoapp;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;
import java.util.List;


public class UsersViewModel extends ViewModel {

    private UserRepositoryRetrofit userRepository;

    public UsersViewModel() {
        userRepository = new UserRepositoryRetrofit();
    }

    public LiveData<List<UserResponse>> getUserResponsesLiveDate() {
        return userRepository.getUserListLiveData();
    }

    public LiveData<String> getUserResponseErrorLiveData() {
        return userRepository.getUserListErrorLiveData();
    }

}

File: app/src/main/java/com/example/retrofitdemoapp/UsersViewModel.kt
package com.example.retrofitdemoapp

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class UsersViewModel : ViewModel() {
    private val userRepository = UserRepositoryRetrofit()
    private val selectedUserMutableLiveData = MutableLiveData<UserResponse>()

    fun getSelectedUserLiveData(): LiveData<UserResponse> = selectedUserMutableLiveData
    fun setSelectedUser(userResponse: UserResponse) {
        selectedUserMutableLiveData.postValue(userResponse)
    }
    fun getUserResponsesLiveData() = userRepository.userListLiveData

    fun getUserResponseErrorLiveData() = userRepository.userListErrorLiveData

}